xfields/beam_elements/beambeam3d.py

```diff
@@ -540,15 +601,15 @@ def _apply_bb_kicks_in_boosted_frame(self, particles):
                 else:
                     return xt.PipelineStatus(on_hold=True)
 
+            # compute interacting other beam slice ID
             self.config_for_update._other_beam_slice_index_for_particles[:] =(
                  self.config_for_update._i_step - self.config_for_update._particles_slice_index)
-            self.config_for_update._other_beam_slice_index_for_particles[
-                             self.config_for_update._particles_slice_index < 0] = -1
+
             self.synchro_beam_kick(particles=particles,
                         i_slice_for_particles=self.config_for_update._other_beam_slice_index_for_particles)
 
             self.config_for_update._i_step += 1
-            if self.config_for_update._i_step == (n_slices_self_beam + self.num_slices_other_beam):
+            if self.config_for_update._i_step == (n_slices_self_beam + self.num_slices_other_beam - 1): # new: -1 
```

Comment:
The beam beam tests on my mac fail because they try to run on GPU but I don't have it. Otherwise one test fails because a missing python module called mako and another one called gpyfft.